teamMaster.py

```diff
@@ -0,0 +1,330 @@
+                                                       'home',
+                                                       daily_team_dict[home])
+
+            daily_team_dict[home].update(daily_team_entry)
+
+            # update away team fields including game history
+            daily_team_entry = update_entry_for_game_2(game_directory,
+                                                       box_level_3,
+                                                       'away',
+                                                       daily_team_dict[away])
+
+            daily_team_dict[away].update(daily_team_entry)
+
+    # return newly updated master entries for all teams playing today
+    return daily_team_dict
+
+
+def invoke_teamMaster_as_sub(gamedate=None):
+    """
+    :param gamedate: date of the games in format "MM-DD-YYYY"
+
+    This routine is invoked when running as imported function vs main driver
+    """
+
+    init_logger()
+    logger.info('Executing script ' + SCRIPT + ' as sub-function')
+    rc = main(gamedate)
+
+    return rc
+
+
+def main(gamedate=None):
+    """
+    main process to update the team master
+    determine file and url names based on date of games
+    load the day's schedule into memory (possible there are no games for date)
+    open current team master and load into memory
+    call function to process the day's schedule, returns updates to master
+    apply updates to team master dictionary, write to file with date in name
+    use shell utility to copy new dictionary to team master file name
+    - this gives us a new master after each udpate and a snapshot at end of day
+    """
+
+    io = determine_filenames(gamedate)
+    schedule_in = io[0]
+    team_out = io[1]
+    date_of_games = io[2]
+
+    # load daily schedule dictionary into memory, must exist or will bypass
+    try:
+        todays_schedule_dict = load_daily_schedule(schedule_in)
+    except LoadDictionaryError:
+        return 20
+
+    logger.info('Creating Team Master file for date: ' + date_of_games)
+
+    # load team master dictionary into memory
+    try:
+        with open(TEAM_MSTR_I, 'r') as teamMaster:
+            team_mstr_dict = json.load(teamMaster)
+    except Exception:
+        team_mstr_dict = {"-file": "Team Master Dictionary File"}
+
+    # use daily schedule to extract info from associated boxscore dictionaries
+    todays_team_dict = process_schedule(todays_schedule_dict, team_mstr_dict)
+
+    # update master dictionary variable and write to snapshot output file
+    team_mstr_dict.update(todays_team_dict)
+
+    with open(team_out, 'w') as teamfile:
+        json.dump(team_mstr_dict, teamfile,
+                  sort_keys=True, indent=4, ensure_ascii=False)
+
+    # copy snapshot to master file name for ongoing updates through the season
+    shutil.copy(team_out, TEAM_MSTR_I)
+
+    return 0
+
+
+if __name__ == '__main__':
+    logging.config.fileConfig(LOGGING_INI)
+    init_logger()
```

Comment:
I see what you're doing in making the logger global, etc., what would you think about passing it in to main()?  I've seen it done both ways but I think initializing it once and passing it is maybe cleaner.